src/__tests__/PartialContainer.spec.ts

```diff
@@ -152,6 +152,43 @@ describe("PartialContainer", () => {
       });
     });
 
+    describe("provide service using provideValue", () => {
+      const dependenciesContainer = Container.provides(Injectable("TestService", () => "old service"));
+
+      describe("and the new Service does not override", () => {
+        const partialContainer = new PartialContainer({}).providesValue("NewTestService", "new service");
+        expect(dependenciesContainer.provides(partialContainer).get("NewTestService")).toEqual("new service");
+      });
+
+      describe("and the new Service does override", () => {
+        const partialContainer = new PartialContainer({}).providesValue("TestService", "new service");
+        expect(dependenciesContainer.provides(partialContainer).get("TestService")).toEqual("new service");
+      });
+    });
+
+    describe("provide service using provideClass", () => {
+      const dependenciesContainer = Container.provides(Injectable("TestService", () => "old service"));
+
+      class NewTestService {
+        static dependencies = ["TestService"] as const;
+        constructor(public testService: string) {}
+      }
+
+      describe("and the new Service does not override", () => {
+        const partialContainer = new PartialContainer({}).providesClass("NewTestService", NewTestService);
+        expect(dependenciesContainer.provides(partialContainer).get("NewTestService")).toBeInstanceOf(NewTestService);
```

Comment:
I think we need an expectation that a dependency is missing, something line:
```ts
const partialContainer = new PartialContainer({}).providesClass("NewTestService", NewTestService);
// @ts-expect-error
expect(() => Container.provides(partialContainer).get("NewTestService")).toThrow(
  '[Container::get] Could not find Service for Token "TestService". ' +
    "This should've caused a compile-time error. If the Token is 'undefined', " +
    "check all your calls to the Injectable function. Make sure you define dependencies " +
    "using string literals or string constants that are definitely initialized before the call to Injectable."
);
```

Would be nice to have that for other `provide*` methods on partial container.